tests/unit/s2n_server_key_share_extension_test.c

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+
+#include <stdint.h>
+#include <s2n.h>
+
+#include "tls/extensions/s2n_server_key_share.h"
+#include "tls/extensions/s2n_client_key_share.h"
+
+#include "testlib/s2n_testlib.h"
+#include "stuffer/s2n_stuffer.h"
+#include "utils/s2n_safety.h"
+
+static int s2n_public_ecc_keys_are_equal(struct s2n_ecc_params *param_1, struct s2n_ecc_params *param_2);
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    /* Test s2n_extensions_server_key_share_send_size */
+    {
+        struct s2n_connection *conn;
+        EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_SERVER));
+        EXPECT_EQUAL(0, s2n_extensions_server_key_share_send_size(conn));
+
+        conn->secure.server_ecc_params.negotiated_curve = &s2n_ecc_supported_curves[0];
+        EXPECT_EQUAL(( 32 * 2 ) + 1 + 8, s2n_extensions_server_key_share_send_size(conn));
```

Comment:
We shouldn't be recalculating share size in this test, since the extension doesn't care what the share size actually is. You can just use negotiated_curve->share_size + 8.